File: Java/MyWork/Sorting/QuickSort.java
package MyWork.Sorting;
import java.util.Arrays;

public class QuickSort {


    static void swap(int arr[], int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    static int partionIndex(int arr[], int start, int end) {
        int pIndex = start;
        int pivot = arr[end];
        for (int i = start; i < end; i++) {
            if (arr[i] < pivot) {
                swap(arr, pIndex, i);
                pIndex++;
            }
        }
        swap(arr, pIndex, end);
        return pIndex;
    }

    static void quickSort(int arr[], int start, int end) {
        if (start < end) {
            int pi = partionIndex(arr, start, end);
            quickSort(arr, start, pi - 1);
            quickSort(arr, pi + 1, end);
        }
    }

    public static void main(String[] args) {
        int[] arr = {11, 2, 3, 5, 23, 64, 9};
        int start = 0, end = arr.length;
        quickSort(arr, start, end -1);
        System.out.println("The array after sorting is : ");
        System.out.println(Arrays.toString(arr));
    }

}

File: CPP/Basics/arrays/sorting/quicksort.cpp
#include<iostream>

using namespace std;

int partitionIndex(int arr[], int low, int high)
{
    int pivot = arr[low];
    int piIndex = 0;
    for(int i=low; i<=high; i++){
        if(arr[i] < pivot)
        {
            int temp = arr[piIndex];
            arr[piIndex++] = arr[i];
            arr[i] = temp;
        }
    }
    return piIndex;
}

void quickSort(int arr[], int low, int high)
{
    if(low<high)
    {
        int pi = partitionIndex(arr, low, high);
        quickSort(arr,low, pi);
        quickSort(arr, pi+1, high);
    }
}

int main()
{
    int n;
    cin >> n;
    int arr[n];
    for(int i=0; i<n; i++)
        cin >> arr[i];
    
    cout << "The elements after sorting are: ";
    quickSort(arr, 0, n-1);
    for(int i=0; i<n; i++)
        cout << arr[i] << " ";
}